cscs-checks/tools/profiling_and_debugging/gdb4hpc.py

```diff
@@ -18,55 +16,64 @@ def __init__(self, lang, extension):
         if lang == 'F90':
             self.build_system.max_concurrency = 1
 
-        self.executable = 'gdb4hpc -v'
+        self.executable = 'gdb4hpc'
+        self.executable_opts = ['-v']
+        self.target_executable = './jacobi'
+        self.gdbcmds = './%s.in' % self.executable
+        self.gdbslm = '%s.slm' % self.executable
+        self.gdbrpt = '%s.rpt' % self.executable
         self.sourcesdir = os.path.join('src', lang)
         self.valid_prog_environs = ['PrgEnv-gnu']
         self.modules = ['gdb4hpc']
-        self.prgenv_flags = ['-g -O2 -fopenmp']
+        self.prgenv_flags = ['-g', '-O2', '-fopenmp']
         self.build_system.cflags = self.prgenv_flags
         self.build_system.cxxflags = self.prgenv_flags
         self.build_system.fflags = self.prgenv_flags
-
         self.num_tasks = 1
         self.num_tasks_per_node = 1
         self.num_cpus_per_task = 4
-        self.num_tasks_per_core= 1
+        self.num_tasks_per_core = 1
         self.num_iterations = 5
         self.variables = {
             'CRAYPE_LINK_TYPE': 'dynamic',
             'OMP_NUM_THREADS': str(self.num_cpus_per_task),
             'ITERATIONS': str(self.num_iterations),
             'OMP_PROC_BIND': 'true',
         }
-        self.maintainers = ['MK', 'JG']
+        self.maintainers = ['JG']
         self.tags = {'production'}
+        self.pre_run = [
```

Comment:
You could a (Python) comment here explaining what is the trick you are doing with the `#GDB4HPC`.